core/services/keystore/eth.go

```diff
@@ -59,6 +60,26 @@ type eth struct {
 	ds            sqlutil.DataSource
 	subscribers   [](chan struct{})
 	subscribersMu *sync.RWMutex
+	resourceMutex map[common.Address]*ResourceMutex // ResourceMutex is an internal field and ought not be persisted to the database. Its main usage is to verify that the same key is not used for both TXMv1 and TXMv2 (usage in both TXMs will cause nonce drift and will lead to missing transactions). This functionality should be removed after we completely switch to TXMv2
+}
+
+// GetResourceMutex gets the resource mutex associates with the address if no resource mutex is found a new one is created
+func (ks *eth) GetResourceMutex(ctx context.Context, address common.Address) (*ResourceMutex, error) {
+	ks.lock.RLock()
+	defer ks.lock.RUnlock()
```

Comment:
Don't you need a write lock since this behaves like a `GetOrCreate`?

Reply:
Fixed in [`72cc299` (#15969)](https://github.com/smartcontractkit/chainlink/pull/15969/commits/72cc299612da2d46234416a5b5999857319f90da)